chore: fix index out of range in metrics metadata (#6035)
grafana/mimir: CHANGELOG.md

* [ENHANCEMENT] Expose `/sync/mutex/wait/total:seconds` Go runtime metric as `go_sync_mutex_wait_total_seconds_total` from all components. #5879
* [ENHANCEMENT] Query-scheduler: improve latency with many concurrent queriers. #5880
* [ENHANCEMENT] Implement support for `limit`, `limit_per_metric` and `metric` parameters for `<Prometheus HTTP prefix>/api/v1/metadata` endpoint. #5890

> * [ENHANCEMENT] Distributor: add experimental support for storing metadata when ingesting metrics via OTLP. This makes metrics description and type available when ingesting metrics via OTLP. Enable with `-distributor.enable-otlp-metadata-storage=true`. #5693 #5957

Reviewer: ```suggestion
* [ENHANCEMENT] Distributor: add experimental support for storing metadata when ingesting metrics via OTLP. This makes metrics description and type available when ingesting metrics via OTLP. Enable with `-distributor.enable-otlp-metadata-storage=true`. #5693 #6035
```

* [ENHANCEMENT] Ingester: added support for sampling errors, which can be enabled by setting `-ingester.error-sample-rate`. This way each error will be logged once in the configured number of times. All the discarded samples will still be tracked by the `cortex_discarded_samples_total` metric. #5584 #6014
* [ENHANCEMENT] Ruler: Fetch secrets used to configure TLS on the Alertmanager client from Vault when `-vault.enabled` is true. #5239
* [BUGFIX] Query-frontend: Don't retry read requests rejected by the ingester due to utilization based read path limiting. #6032

> * [BUGFIX] Distributor: fix index out of range introduced in experimental feature from #5693. #5957

Reviewer 2: ```suggestion
* [BUGFIX] Distributor: fix index out of range introduced in OTLP Metadata handling (from #5693). #5957
```

> * [BUGFIX] Distributor: fix index out of range introduced in OTLP Metadata handling (from #5693). #5957

Reviewer 3: The change should refer to the PR.
```suggestion
* [BUGFIX] Distributor: fix index out of range introduced in OTLP Metadata handling (from #5693). #6035 
```

Reviewer: Just add the PR number to the same CHANGELOG entry of `#5693`, given it was not part of a release yet.

Author: Done
